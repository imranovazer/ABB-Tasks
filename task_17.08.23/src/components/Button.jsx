import React from "react";

const Button = ({ backgroundColor, text, onClick }) => {
  return (
    <button
      className="button"
      style={{ backgroundColor: backgroundColor || "grey" }}
      onClick={onClick}
    >
      {text}
    </button>
  );
};

export default Button;
