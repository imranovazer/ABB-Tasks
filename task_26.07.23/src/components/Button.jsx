import React from "react";

class Button extends React.Component {
  render() {
    const { backgroundColor, text, onClick } = this.props;

    return (
      <button
        className="button"
        style={{ backgroundColor: backgroundColor || "grey" }}
        onClick={onClick}
      >
        {text}
      </button>
    );
  }
}

export default Button;
