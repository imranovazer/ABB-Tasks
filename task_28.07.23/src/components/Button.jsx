import React, { Component } from "react";

class Button extends Component {
  render() {
    const { backgroundColor, text, onClick } = this.props;

    return (
      <button
        className="button"
        style={{ backgroundColor: backgroundColor || "grey" }}
        onClick={onClick}
      >
        {text}
      </button>
    );
  }
}

export default Button;
