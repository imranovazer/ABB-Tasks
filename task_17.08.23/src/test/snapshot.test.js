import React from 'react';
import { render } from '@testing-library/react';

import Button from '../components/Button';


test('renders Button correctly', () => {
    const { asFragment } = render(<Button />);


    expect(asFragment()).toMatchSnapshot();
});
