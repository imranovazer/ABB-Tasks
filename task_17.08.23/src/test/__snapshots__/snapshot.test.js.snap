// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button correctly 1`] = `
<DocumentFragment>
  <button
    class="button"
    style="background-color: grey;"
  />
</DocumentFragment>
`;
